src/server/game/AI/SmartScripts/SmartScriptMgr.cpp

```diff
@@ -2123,6 +2124,19 @@ bool SmartAIMgr::IsEventValid(SmartScriptHolder& e)
             TC_SAI_IS_BOOLEAN_VALID(e, e.action.setUninteractible.uninteractible);
             break;
         }
+        case SMART_ACTION_ACTIVATE_GAMEOBJECT:
+        {
+            if (!NotNULL(e, e.action.activateGameObject.gameObjectAction))
+                return false;
+
+            if (e.action.activateGameObject.gameObjectAction >= uint32(GameObjectActions::Max))
+            {
+                TC_LOG_ERROR("sql.sql", "SmartAIMgr: Entry %d SourceType %u Event %u Action %u has gameObjectAction parameter out of range (max allowed %u, current value %u), skipped.",
+                    e.entryOrGuid, e.GetScriptType(), e.event_id, e.GetActionType(), uint32(GameObjectActions::Max), e.action.activateGameObject.gameObjectAction);
```

Comment:
Max allowed is Max - 1 here, not Max

Reply:
Why it's - 1?

Reply:
Yeah, I see
I forgot I removed your Max = SetTapList
Then uint32(GameObjectActions::SetTapList) or it's not what you wanted for both branches?

Reply:
We should still use Max, just writing `uint32(Max) - 1` in the log.
I'll change it when I find time :)